File: 1218-longest-arithmetic-subsequence-of-given-difference/1218-longest-arithmetic-subsequence-of-given-difference.go
func longestSubsequence(arr []int, difference int) int {
    
    dp := make(map[int]int)
    
    var answer int = 0
    
    for _, number := range arr{
        value, flag := dp[number - difference]
        
        if flag{
            dp[number] = value + 1
        }else{
            dp[number]  = 1
        }
        
        if answer < dp[number]{
            answer = dp[number]
        }
    }
    return answer
}

File: 1218-longest-arithmetic-subsequence-of-given-difference/1218-longest-arithmetic-subsequence-of-given-difference.py
class Solution:
    def longestSubsequence(self, arr: List[int], difference: int) -> int:
        
        dp = defaultdict(int)
        result = 0
        
        for val in arr:
            
            if val - difference in dp:
                dp[val] = dp[val - difference] + 1
            
            else:
                dp[val] = 1
            
            result = max(result, dp[val])
        
        return result